mathesar/api/viewsets/db_types.py

```diff
@@ -0,0 +1,18 @@
+from rest_framework import viewsets
+from rest_framework.response import Response
+from db.types.base import db_types_hinted, PostgresType, MathesarCustomType
+from itertools import chain
+
+
+class DbTypeViewSet(viewsets.ViewSet):
+    def list(self, _):
+        known_vanilla_types = (postgres_type for postgres_type in PostgresType)
+        known_custom_types = (mathesar_custom_type for mathesar_custom_type in MathesarCustomType)
+        known_db_types = chain(known_vanilla_types, known_custom_types)
+        data = [
```

Comment:
Please use a serializer to generate this, this logic does not belong in the viewset.

Reply:
Also, you will need to specify a list of databases for every type. Mathesar might be connected to multiple databases and not all of them will have every type installed. We need to check what types are installed where.

Reply:
It would be ideal to support filtering the types endpoint by database ID so that if a client is working with a single database, they can filter to types available in that database. I think we have this implemented for schemas already so you can reference that.